scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+                ;
+            recordManager.commit();
+            if (!firstAttempt) {
+                logger.info("Loading of job database successful after clean.");
+            }
+        } catch (Throwable e) {
+            if (firstAttempt) {
+                logger.error("Error occurred when loading channels database " + statusFile.getAbsolutePath() +
+                             ", now cleaning it and retrying.", e);
+                cleanDataBase();
+                init(false);
+            } else {
+                close();
+                throw new IllegalStateException("Error when loading database (even after cleaning it): " +
+                                                statusFile.getAbsolutePath(), e);
+            }
+        }
+        return true;
+    }
+
+    public boolean isStarted() {
+        return true;
+    }
+
+    public boolean cleanDataBase() {
+        close();
+        logger.info("Cleaning database");
+
+        // delete all db files
+        File[] dbChannelFiles = statusFileDirectory.listFiles((dir,
+                name) -> name.startsWith(DEFAULT_STORE_SESSION_NAME));
+        for (File file : dbChannelFiles) {
+            try {
+                logger.info("Deleting " + file);
+                file.delete();
+            } catch (Exception e) {
+                logger.info("Error while deleting file during database cleanup", e);
+            }
+        }
+        return true;
+    }
+
+    public boolean close() {
+        inMemoryChannels.clear();
+        if (recordManager != null) {
+            try {
+                logger.info("Closing Record Manager");
+                recordManager.close();
+                recordManager = null;
+            } catch (IOException e) {
+                // ignored
+            }
+        }
+        return true;
+    }
+
+    private Channel getChannel(String name) throws InvalidChannelException {
+        if (inMemoryChannels.containsKey(name)) {
+            return inMemoryChannels.get(name);
+        } else if (persistedChannels.containsKey(name)) {
+            return persistedChannels.get(name);
+        }
+        throw new InvalidChannelException("Channel " + name + " does not exist");
+    }
+
+    @Override
+    public boolean createChannel(String originator, TaskId taskid, String name, boolean isPersistent)
+            throws IOException {
+        try {
+            boolean alreadyExistingChannel = deleteChannel(originator, taskid, name);
+            Channel newChannel = new Channel();
+            if (isPersistent) {
+                persistedChannels.put(name, newChannel);
+                logWithContext(taskid, null, "Created new persistent channel " + QUOTE + name + QUOTE, Level.INFO);
+                recordManager.commit();
+            } else {
+                logWithContext(taskid, null, "Created new memory channel " + QUOTE + name + QUOTE, Level.INFO);
+                inMemoryChannels.put(name, newChannel);
```

Comment:
The order in which
- log is written
- channel is put
is different in the two if/else clauses. We should use the same order.